cmd/sync/runner.go

```diff
@@ -181,15 +184,26 @@ func (r *runner) sync(ctx context.Context, srcRegistry, dstRegistry registry.Int
 	var err error
 
 	fmt.Println()
-	fmt.Printf("Logging in destination registry...\n")
-	err = dstRegistry.Login(ctx, r.flag.DstRegistryUser, r.flag.DstRegistryPassword)
-	if err != nil {
-		return microerror.Mask(err)
+
+	if r.lastLoginAt == nil {
+		fmt.Printf("Logging in destination registry...\n")
+		err = dstRegistry.Login(ctx, r.flag.DstRegistryUser, r.flag.DstRegistryPassword)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		timeNow := time.Now()
+		r.lastLoginAt = &timeNow
+	} else {
+		fmt.Printf("Already logged in\n")
 	}
+
 	defer func(ctx context.Context) {
-		fmt.Println()
-		fmt.Printf("Logging out of destination registry...\n")
-		_ = dstRegistry.Logout(ctx)
+		if r.lastLoginAt != nil && time.Since(*r.lastLoginAt) >= loginTTL {
+			fmt.Println()
```

Comment:
Did you check how messages look like after moving those Printlns?

Reply:
They're ok now.